_posts/2020-05-18-websocket-fallback-sockjs.md

```diff
@@ -0,0 +1,329 @@
+
+## Praktyka
+Ostatecznie został wybrany SockJS z powodu prostoty wykorzystania go w istniejącym springowym projekcie.
+
+### Wersja przed poprawkami z użyciem czystego WebSocket
+#### Backend
+* pom.xml
+{% highlight xml %}
+<dependencies>
+    <dependency>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-websocket</artifactId>
+    </dependency>
+</dependencies>
+{% endhighlight %}
+
+* WebSocketConfig.java
+{% highlight java %}
+@Configuration
+@EnableWebSocket
+public class WebSocketConfig implements WebSocketConfigurer {
+    private SocketHandler socketHandler;
+
+    public WebSocketConfig(SocketHandler socketHandler) {
+        this.socketHandler = socketHandler;
+    }
+
+    @Override
+    public void registerWebSocketHandlers(WebSocketHandlerRegistry registry) {
+        registry.addHandler(socketHandler, "/signaling");
+    }
+}
+{% endhighlight %}
+
+#### Frontend
+* web-socket-manager.ts
+{% highlight typescript %}
+export class WebSocketManager {
+    private webSocket?: WebSocket;
+    constructor() {
+        this.webSocket = this.initWebSocket();
+    }
+     
+    initWebSocket(): WebSocket {
+        return new WebSocket('https://localhost/signaling');
+    }
+}
+{% endhighlight %}
+
+
+### Wersja po poprawkach z użyciem SockJS
+* pom.xml (bez zmian)
+{% highlight diff %}
+<dependencies>
+    <dependency>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-websocket</artifactId>
+    </dependency>
+</dependencies>
+{% endhighlight %}
+
+* WebSocketConfig.java
+{% highlight diff %}
+@Configuration
+@EnableWebSocket
+public class WebSocketConfig implements WebSocketConfigurer {
+    private SocketHandler socketHandler;
+
+    public WebSocketConfig(SocketHandler socketHandler) {
+        this.socketHandler = socketHandler;
+    }
+
+    @Override
+    public void registerWebSocketHandlers(WebSocketHandlerRegistry registry) {
+-       registry.addHandler(socketHandler, "/signaling");
++       registry.addHandler(socketHandler, "/signaling")
++           .withSockJS();
+    }
+}
+{% endhighlight %}
+
+* web-socket-manager.ts
+{% highlight diff %}
++import * as SockJS from 'sockjs-client';
++
+ export class WebSocketManager {
+    private webSocket?: WebSocket;
+    constructor() {
+        this.webSocket = this.initWebSocket();
+    }
+    
+    initWebSocket(): WebSocket {
+-      return new WebSocket('https://localhost/signaling');
++      return new SockJS('https://localhost/signaling');
+    }
+ }
+{% endhighlight %}
+
+I teoretycznie to wystarczy do działania, chociaż w celu zostawienia kompatybilności wstecz 
+oraz rozwiązania problemów podczas których wykorzystujemy SockJS w bibliotece wybudowanej za pomocą RollUp,
+konieczne jest zmodyfikowanie kodu. 
+
+W przypadku kompatybilności wstecz, został dodany dodatkowy endpoint wykorzystujący wersjonowanie.
```

Comment:
Jaki problem z rollup? Co konkretnie?
Nie pisałbym o tym. To jest jakaś ułomność naszego procesu budowania lub naszej konfiguracji. Nie ma to nic wspólnego z samym SockJS i jedynie wprowadza niepotrzebny szum do wpisu.